Mage/src/main/java/mage/abilities/keyword/MutateAbility.java

```diff
@@ -1,39 +1,191 @@
+
+    public MutateAbility(Card card, String mutateCost) {
+        this(card, createCosts(new ManaCostsImpl<>(mutateCost)));
+    }
+
+    public MutateAbility(Card card, Costs<Cost> mutateCosts) {
+        super(Zone.ALL, null);
+        this.mutateCosts = mutateCosts;
+        this.name = MUTATE_KEYWORD;
+        this.activated = false;
+
+        StringBuilder sb = new StringBuilder();
+        sb.append(MUTATE_KEYWORD).append(' ');
+        for (Cost cost : mutateCosts) {
+            if (!(cost instanceof ManaCosts)) {
+                sb.setLength(sb.length() - 1);
+                sb.append("&mdash;");
+                break;
+            }
+        }
+        sb.append(mutateCosts.getText());
+        if (!(mutateCosts.get(mutateCosts.size() - 1) instanceof ManaCosts)) {
+            sb.append('.');
+        }
+        sb.append(' ').append(MUTATE_REMINDER);
+
+        this.ruleText = sb.toString();
     }
 
     private MutateAbility(final MutateAbility ability) {
         super(ability);
+        this.ruleText = ability.ruleText;
+        this.mutateCosts = ability.mutateCosts;
+    }
+
+    private static Costs<Cost> createCosts(Cost cost) {
+        Costs<Cost> costs = new CostsImpl<>();
+        costs.add(cost);
+        return costs;
     }
 
     @Override
     public MutateAbility copy() {
         return new MutateAbility(this);
     }
 
+    @Override
+    public boolean isActivated(Ability source, Game game) {
+        return activated;
+    }
+
+    @Override
+    public String getCastMessageSuffix(Game game) {
+        return " with " + mutateCosts.getText();
+    }
+
+    @Override
+    public boolean isAvailable(Ability source, Game game) {
+        return source.getZone().equals(Zone.HAND) || source.getZone().equals(Zone.COMMAND);
+    }
+
+    @Override
+    public boolean askToActivateAlternativeCosts(Ability ability, Game game) {
+        if (ability.getAbilityType() == AbilityType.SPELL) {
+            Player player = game.getPlayer(ability.getControllerId());
+            Spell spell = game.getStack().getSpell(ability.getId());
+            if (player != null && spell != null) {
+                if (mutateCosts.canPay(ability, this, ability.getControllerId(), game)) {
+                    if (player.chooseUse(Outcome.Benefit,
+                            "Cast " + spell.getLogName() + " with mutate for " + getCosts().getText() + " ?", ability, game)) {
+                        mutateCosts.clearPaid();
+                        TargetCreaturePermanent target = new TargetCreaturePermanent(filter);
+                        target.setNotTarget(true);
+                        if (target.canChoose(getSourceId(), player.getId(), game)) {
+                            if (player.chooseTarget(Outcome.Benefit, target, ability, game)) {
+                                Permanent targetPermanent = game.getPermanent(target.getFirstTarget());
+                                boolean mutateUnder = player.chooseUse(Outcome.Neutral,
```

Comment:
"mutate under" or "mutate over" are not "modes" nor are they choices you make as you cast the spell. You choose over or under as the spell resolves.

Reply:
This is something I was going over when thinking about copying a mutate spell (like with Volo), and I thought I read somewhere that every aspect of the spell needs to be the same, which I thought included the over/under. However, I can't find anything saying that in the rules, so I'm pretty sure you're right.

Reply:
> 702.140c As a mutating creature spell resolves, if its target is legal, it doesn’t enter the battlefield. Rather, it merges with the target creature and becomes one object represented by more than one card or token (see rule 723, “Merging with Permanents”). The spell’s controller chooses whether the spell is put on top of the creature or on the bottom. The resulting permanent is a mutated permanent.

Reply:
How does it currently work for a card like [[Momentary Blink]] or a creature with persist? Any time a mutated permanent leaves and re-enters it needs to come back as each individual creature separately.

Reply:
> How does it currently work for a card like [[Momentary Blink]] or a creature with persist? Any time a mutated permanent leaves and re-enters it needs to come back as each individual creature separately.

Momentary Blink should be covered by my `setupTestMutateToExileAndBack` unit tests, however, Persist and Regenerate seem to need more work to implement properly.

Reply:
> Momentary Blink should be covered by my `setupTestMutateToExileAndBack` unit tests, however, Persist and Regenerate seem to need more work to implement properly.

Persist, Undying, and Regenerate are all working now